rfcs/0001-shabka-usage-design.md

```diff
@@ -0,0 +1,186 @@
+is the host's release (defined in `.hosts/hosts/{host}/.release`) if
+defined, or `shabka`'s default, defined in `shabka/.release`.
+
+> This command cannot build a host with a different kernel than the host it is
+> being ran on, e.g. a `Darwin` host cannot be built on a `Linux` host.
+
+This command takes care of setting all environment variables and options to
+ensure a smooth `nix-build` depending on what is defined in `shabka`, `.shabka`
+and `dotshabka-user`. The default locations of `.shabka` and `dotshabka-user`
+are defined in their respective section of this document.
+
+### `shabka diff`
+
+This will do what `scripts/diff-host.sh` used to do. Its usage is:
+
+```sh
+shabka diff <host> <base-rev> [target-rev]
+```
+
+By default the `target-rev` argument is HEAD.
+
+/!\ This section needs improvement given the fact that we now have 2
+repositories to deal with (`shabka` and `.shabka`, as `dotshabka-user` is
+defined in `.shabka`). This implies a lot of arguments and some designing
+around this command must be done.
+
+### `shabka push-to-cachix`
+
+This will do what `scripts/push-to-cachix.sh` currently does. Its usage is:
+
+```sh
+shabka push-to-cachix <cache-name> [host1] [host2] [host3] [...]
+```
+
+If no host is given, this command will push every host in `.shabka/hosts`. This
+command requires the environment variable `CACHIX_SIGNING_KEY` to be set.
+
+## `dotshabka-user` for a user configuration
+
+This directory will be used to define a user's personal configuration.
+
+This directory must have a `home.nix` that defines a
+function for the user as its now done in the `home.nix` inside a host's
+configuration. In this `home.nix`, the user can then define its personal
+configuration, using the `home` module in `shabka`, overriding it, or directly
+using `home-manager` configuration options.
+
+There must also be a `default.nix` inside this directory, which imports
+`home.nix` and can then be imported by a `.shabka` repository to be used as a
+user's configuration.
+
+This `default.nix` will later be imported inside a host's configuration, when
+defining `shabka.users.users`, by assigning one function per user. This allows
+for the user of the same host to have different configurations. Also, it makes
+it easy for a user's configuration to be imported in several `.shabka`, and even
+be used outside of the `shabka` use case.
+
+## `.shabka` for hosts configuration
+
+By default, `shabka` will look for this directory in `$HOME/.config/shabka` and
+`$HOME/.shabka`, in this order. This should be configurable in some way. Here
+is the expected structure of the directory:
+
+* `external/`: Nix expressions for fetching externals such as a personal NUR,
+  or a list of your SSH keys.
+* `hosts/`: top-level expressions specific to individual hosts. The name of the
+  host's folder must match its hostname.
+
+Inside a host's directory, `shabka` will expect the following things:
+
+* `.uname`: either containing `Darwin` or `Linux`
```

Comment:
I think the `.release` and the `.uname` names are poor design decisions on my part. I don't think there's value in hiding this file, in fact, I think it's counterproductive. Plus the file itself is already in a hidden directory. M

Reply:
How do you propose to handle this?

Reply:
I propose just using `release` and `uname` instead of `.release` and `.uname`.